src/BridgeIHMCORS.cpp

```diff
@@ -19,6 +19,77 @@ namespace yarp
 namespace dev
 {
 
+BridgeIHMCORSReceiverThread::BridgeIHMCORSReceiverThread(): m_parentDevice(nullptr),
+                                                            m_desiredSocket(nullptr),
+                                                            m_desiredService(nullptr)
+{}
+
+bool BridgeIHMCORSReceiverThread::configure(BridgeIHMCORS *parentDevice, yarp::os::Searchable& config)
+{
+    m_parentDevice = parentDevice;
+
+    // Configure inbound socket (receiving commands)
+    yarp::os::Value desiredPortNumber = config.check("desired-port-number", yarp::os::Value("9980"));
+    yarp::os::Value desiredAddressVal = config.find("desired-address");
+
+    if (desiredAddressVal.isNull())
+    {
+        yError() << "IP address for receiving data not found";
+        return false;
+    }
+
+    // Trying to match the specified IP address and port number to an endpoint
+    m_desiredService = new asio::io_service;
+    asio::ip::udp::resolver resolverDesired(*m_desiredService);
+    asio::ip::udp::resolver::query queryDesired(asio::ip::udp::v4(), desiredAddressVal.asString(), desiredPortNumber.asString());
+    m_desiredEndpoint = *resolverDesired.resolve(queryDesired);
+
+    // TODO(nava): substitute this with an error message in case the parameter is not found
+    std::cout << "Current endpoint for receiving data:" << m_desiredEndpoint << std::endl;
+
+    m_desiredSocket = new asio::ip::udp::socket(*m_desiredService);
+
+    if (!m_desiredSocket)
+    {
+        yError() << "Fail to create the ""desired"" socket";
+        return false;
+    }
+    m_desiredSocket->open(asio::ip::udp::v4());
+    m_desiredSocket->bind(m_desiredEndpoint);
+
+    return true;
+}
+
+void BridgeIHMCORSReceiverThread::resize(const int nrOfJoints)
+{
+    m_robotDesired.jointDesireds().resize(nrOfJoints);
+    m_desiredBuffer.resize(m_robotDesired.getMaxCdrSerializedSize());
```

Comment:
`getMaxCdrSerializedSize` returns a very big number, independent from the number of joints.

Reply:
Good catch, changing it to `getCdrSerializedSize` .